tests.js

```diff
@@ -86,13 +117,35 @@ QUnit.module("Logo Unit Tests", {
     };
 
     this.assert_predicate = function(expression, predicate) {
-      t.ok(predicate(this.interpreter.run(expression, {returnResult: true})), expression);
+      var result = this.interpreter.run(expression, {returnResult: true});
+      if (result && result.then) {
```

Comment:
This would be much simpler/easier to read if always async. So wrap in a Promise, or preferably just make interpreter.run return a Promise